utils/utils.go

```diff
@@ -0,0 +1,42 @@
+package utils
+
+import (
+	"fmt"
+
+	"github.com/manifoldco/promptui"
+	"github.com/rs/zerolog/log"
+)
+
+func GetTelemetryConsent() bool {
```

Comment:
Yes, it is intended behaviour

Reply:
This would render the cli useless in integrations like jenkins. Jenkins might execute `crda version` for debugging purpose.

Reply:
Its one-time activity no..? any way in Jenkins, admin can override this behaviour by manually setting `consent_telemetry` variable in env

Reply:
In any cli/program I used till now, consent is asked in starting only, no matter what command you execute. I feel this behaviour should stay as is.

Reply:
>Its one-time activity no..? any way in Jenkins, admin can override this behaviour by manually setting consent_telemetry variable in env

I don't think the jenkins plugin deals with the config anyways? do you mean env variable?
CC: @yzainee

Reply:
prompting this in crda version will break the jenkins flow. expectation is to get a version as a response. I dont feel having this in version command is a good idea.
@arajkumar to answer your question, this can be done via secrets only as a persistent data.
While calling crda analyse, the user will have an option to send extra config variables and all, but the version command is internally used. User will have no ways of sending any value here

Reply:
But we need to think about where to provide these statement prompts. we cannot spray it all over the places.

Reply:
> > Its one-time activity no..? any way in Jenkins, admin can override this behaviour by manually setting consent_telemetry variable in env
> 
> I don't think the jenkins plugin deals with the config anyways? do you mean env variable?
> CC: @yzainee

@arajkumar By Env variables, I mean parallel to `crda_key`, which is already in Jenkins Space,  add one more `consent_telemetry` = false/true` to override this Consent Prompt. Nothing will break.

Reply:
value for crda key comes from secrets. as i already mentioned. Do we want to store telemetry consent as a secret?

Reply:
Also, are you tracking any action when a use is checking the version of the crda cli? if no, then why are you asking him the consent to track in this command?

Reply:
@yzainee In Any way, you have to set `consent_telemetry` in Jenkins Env, be it in secrets or anywhere else to use next release in Jenkins space. You can't use further commands `crda analyse` without it. Prompt is just a way to set it in an automated fashion.

Now I believe, if that is the requirements, you can't bypass, you there should not be any problem in setting it in the first place. Set it and use CLI Freely. There will be No prompt to bother you.

Reply:
please re - read: crda version is an internal command. user is not exposed to it.
Having said that, i said it doesnt make sense to have the telemetry on/off for version command.

Coming back to original question posted by @arajkumar , we either need to pick secrets or a have a way to send this value while calling the command and give a disclaimer that if no value is provided, it will be considered false.
What do you think @arajkumar @prashbnair

Reply:
> Question to @deepak1725 : Suppose i send consent_telemetry = true for first 5 calls and then send consent_telemetry = false from 6th onwards, it will stop tracking from 6th onwards. right?
 
Yes. We will stop tracking as soon as consent is set to false

Reply:
> please re - read: crda version is an internal command. user is not exposed to it.
> Having said that, I said it doesn't make sense to have the telemetry on/off for version command.

I think there's been a disconnect here.

By internal, I am sure there is a config file associated with that too. (CLI can't work without config). AFAIK in Jenkins, Env Variables are shared globally.. so ideally your internal commands should have access to Globally set variables Idea is to add `consent_telemetry`  one of it.

Reply:
I have already mentioned about secrets & config values. So providing value is not a concern. 

Question here is why are you asking for telemetry question for version command? it should not be prompted when somebody runs a version cmd. This question has no relation to tekton, jenkins, cli or gh actions.

Reply:
long story short: telemetry prompt should appear for crda analyse

Reply:
@deepak1725 @arajkumar @prashbnair what was the conclusion on this? i see that its already merged